frontend/src/app/signUp/Organization/OrganizationForm.tsx

```diff
@@ -0,0 +1,223 @@
+  }
+
+  const getFormElement = (
+    field: keyof OrganizationCreateRequest,
+    label: string | React.ReactNode,
+    required: boolean,
+    subheader: string | null
+  ) => {
+    switch (field) {
+      case "state":
+        return (
+          <Select
+            label={label}
+            name="state"
+            value={organization.state || ""}
+            options={stateCodes.map((c) => ({ label: c, value: c }))}
+            defaultSelect
+            onChange={onDetailChange("state")}
+            onBlur={() => {
+              validateField("state");
+            }}
+            validationStatus={getValidationStatus("state")}
+            errorMessage={errors.state}
+            required
+          />
+        );
+      case "type":
+        return (
+          <Select
+            label={label}
+            name="type"
+            value={organization.type || ""}
+            options={Object.entries(
+              OrganizationTypeEnum
+            ).map(([key, value]) => ({ label: value, value: key }))}
+            defaultSelect
+            onChange={onDetailChange("type")}
+            onBlur={() => {
+              validateField("type");
+            }}
+            validationStatus={getValidationStatus("type")}
+            errorMessage={errors.type}
+            required
+          />
+        );
+      default:
+        // This is not a field, just some explanatory text
+        if (label === "Organization administrator") {
+          return (
+            <>
+              <label className="usa-label margin-top-3">
+                <strong>{label}</strong>
+                <TextWithTooltip
+                  tooltip="Can be anyone that works at your organization. Adds testing facility locations and manages the organization account."
+                  text="What's an organization administrator?"
+                />
+              </label>
+              <p className="usa-hint">
+                Only one person from an organization can be the administrator.
+                This person will submit information for identity verifcation.
+              </p>
+            </>
+          );
+        }
+        return (
+          <Input
+            className={subheader ? "margin-top-0" : ""}
+            label={label}
+            type={"text"}
+            field={field}
+            key={field}
+            formObject={organization}
+            onChange={onDetailChange}
+            errors={errors}
+            validate={validateField}
+            getValidationStatus={getValidationStatus}
+            required={required}
+            hintText={subheader ? subheader : undefined}
+          />
+        );
+    }
+  };
+
+  return (
+    <CardBackground>
+      <Card logo bodyKicker="Sign up for SimpleReport">
+        <div className="margin-bottom-2 organization-form">
+          {Object.entries(organizationFields).map(
+            ([key, { label, required, subheader }]) => {
```

Comment:
**praise:** love this, it's so concise!

Reply:
Oh thank you! It saves a lot of lines but I also feel like I could document it better for other devs, so I'm going to add some comments in here before I merge

Reply:
Yeah some additional documentation would probably be good - I followed it but only because I was doing a close review of the lines above, haha. Thanks!

Reply:
Added some comments!